js_modules/dagit/packages/core/src/overview/OverviewAssetsRoot.tsx

```diff
@@ -0,0 +1,433 @@
+                    }}
+                  />
+                  {statuses.failed.length}
+                </Box>
+              </Tag>
+            </SelectOnHover>
+          ) : (
+            0
+          )}
+        </Cell>
+        <Cell isLoading={!!statuses.loading}>
+          {statuses.inprogress.length ? (
+            <SelectOnHover assets={statuses.inprogress}>
+              <Tag intent="primary" icon="spinner">
+                {statuses.inprogress.length}
+              </Tag>
+            </SelectOnHover>
+          ) : (
+            0
+          )}
+        </Cell>
+        <Cell isLoading={!!statuses.loading}>
+          {statuses.successful.length ? (
+            <SelectOnHover assets={statuses.successful}>
+              <Tag intent="success">
+                <Box flex={{direction: 'row', alignItems: 'center', gap: 6}}>
+                  <div
+                    style={{
+                      backgroundColor: Colors.Green500,
+                      width: '10px',
+                      height: '10px',
+                      borderRadius: '50%',
+                    }}
+                  />
+                  {statuses.successful.length}
+                </Box>
+              </Tag>
+            </SelectOnHover>
+          ) : (
+            0
+          )}
+        </Cell>
+      </RowGrid>
+    </Row>
+  );
+}
+
+const RowGrid = styled(Box)`
+  display: grid;
+  grid-template-columns: ${TEMPLATE_COLUMNS};
+  height: 100%;
+  > * {
+    padding-top: 26px 0px;
+  }
+`;
+
+const Cell = ({children, isLoading}: {children: React.ReactNode; isLoading?: boolean}) => {
+  return (
+    <RowCell style={{color: Colors.Gray900}}>
+      {isLoading ? (
+        <Box flex={{justifyContent: 'center', alignItems: 'center'}} style={{height: '82px'}}>
+          <Spinner purpose="body-text" />
+        </Box>
+      ) : (
+        <Box flex={{direction: 'row', alignItems: 'center', grow: 1}}>{children}</Box>
+      )}
+    </RowCell>
+  );
+};
+
+const RepositoryLinkWrapper = styled.div<{maxWidth?: number}>`
+  font-size: 12px;
+  pointer-events: none;
+  a {
+    color: ${Colors.Gray600};
+    pointer-events: none;
+    max-width: ${({maxWidth}) => (maxWidth ? 'unset' : `${maxWidth}px`)};
+  }
+`;
+
+function SelectOnHover({assets, children}: {assets: Assets; children: React.ReactNode}) {
+  return (
+    <SelectWrapper>
+      <Select
+        items={assets}
+        itemPredicate={(query, item) =>
+          displayNameForAssetKey(item.key).toLocaleLowerCase().includes(query.toLocaleLowerCase())
+        }
+        itemRenderer={(item) => (
+          <Link to={assetDetailsPathForKey(item.key)} target="_blank">
+            <MenuItem
+              key={displayNameForAssetKey(item.key)}
+              text={displayNameForAssetKey(item.key)}
+              icon="open_in_new"
+            />
+          </Link>
+        )}
+        onItemSelect={() => {}}
+      >
+        {children}
+      </Select>
+    </SelectWrapper>
+  );
+}
```

Comment:
This is interesting, not sure we've had an interaction quite like this anywhere else